app/main.py

```diff
@@ -1,25 +1,62 @@
+from __future__ import annotations
 from abc import ABC
 
 
 class IntegerRange:
-    pass
+    def __init__(self, min_amount: int, max_amount: int) -> None:
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+
+    def __set_name__(self, owner: type, name: str) -> None:
+        self.protected_name = "_" + name
+
+    def __get__(self, instance: type, owner: type) -> str:
+        return getattr(instance, self.protected_name)
+
+    def __set__(self, instance: type, value: int) -> None:
+        if not isinstance(value, int):
+            raise TypeError
+        if not (self.min_amount <= value <= self.max_amount):
+            raise ValueError
```

Comment:
This line raises a `ValueError` without a message. Consider adding a message to provide more context about the error.